### PR TITLE
lab/etapele-compilarii: Ex 6 Changed Makefile

### PR DESCRIPTION
**Pull request by robertZamfir601**
On ex 6:
- Changed Makefile
- Ex 6 requires the students to call functions from 'shop.o' interface
- The old Makefile is a generic one that (at least in my case) doesn't link with 'shop.o'
- The one I added has a 'To Do' for students to complete it, so it links with 'shop.o'
- It will make them look out for Makefile properties or ask

### CODE REVIEW

File: laborator/content/etapele-compilarii/06-obj-link-dev/Makefile [hunk 1]
```diff
@@ -1,2 +1,14 @@
...
+
+all: main
+
+# ** TO DO **
+# Add dependecies here so that the Makefile works
+main: main.c
+	$(CC) $(CFLAGS) -o $@ $^
+
+clean:
+	-rm -f main main.o
```
Comment (teodutu) [inline laborator/content/etapele-compilarii/06-obj-link-dev/Makefile:14]:
```suggestion
	-rm -f main main.o
```
`checkpatch.pl` was also complaining about this. Always check out these errors when you create PRs. To silence old ones, squash your commits into a single one and then `git push -f`.

File: laborator/content/etapele-compilarii/06-obj-link-dev/Makefile [hunk 2]
```diff
@@ -1,2 +1,14 @@
...
-include ../../utils/Makefile.generic
+CC = gcc
+CFLAGS = -g -Wall -Wextra -Werror -fno-pic -masm=intel
+
+.PHONY: all clean
+
+all: main
+
+# ** TO DO **
+# Add dependecies here so that the Makefile works
```
Comment (teodutu) [inline laborator/content/etapele-compilarii/06-obj-link-dev/Makefile:9]:
```suggestion
# TODO: Add dependecies here so that the Makefile works
```

**Review by teodutu — CHANGES_REQUESTED** [review]
I like your solution. You need to modify the `README.md` file and tell students that they are supposed to modify the makefile as well.

In addition to my inline comments, in the future don't make PRs from branches in this repo. This pollutes the repo with useless branches. Use your fork, like you did here: https://github.com/systems-cs-pub-ro/iocla/pull/147 (which needs work btw).

File: laborator/content/etapele-compilarii/06-obj-link-dev/Makefile (hunk 2 above)
Comment (robertZamfir601) [reply laborator/content/etapele-compilarii/06-obj-link-dev/Makefile]: The commit you are referring it is not mine, I always used this workflow. I will change it for the next commits.

File: laborator/content/etapele-compilarii/README.md [hunk 3]
```diff
@@ -443,6 +443,9 @@ Editați fișierul `main.c` pentru a apela corespunzător interfața expusă și
 price is 21
 quantity is 42
 ```
+> **NOTE:**
+> Pentru a putea folosi comanda `make` la acest exercițiu va trebui să completați fișierul `Makefile` pentru a realiza linkarea cu interfața
+> `shop.o` (trebuie tratată ca o bibliotecă externă)
```
Comment (teodutu) [inline laborator/content/etapele-compilarii/README.md:448]:
```suggestion
> Pentru a putea folosi comanda `make` la acest exercițiu va trebui să completați fișierul `Makefile` pentru a realiza linkarea cu fișierul obiect
> `shop.o`
```
If you were to treat it as a library, you would have to add `-L. -lname` etc. It's just an object file.

**Review by teodutu — CHANGES_REQUESTED** [review]
Nearly there. The explanation in the `README.md` is a bit off.

**Review by teodutu — APPROVED** [review]
:+1:

**Pull request merged**